src/librustdoc/passes/collect_intra_doc_links.rs

```diff
@@ -1077,6 +1118,78 @@ impl LinkCollector<'_, '_> {
         }
     }
 
+    fn resolve_with_disambiguator_cached(
+        &mut self,
+        disambiguator: Option<Disambiguator>,
+        item: &mut Item,
+        dox: &str,
+        path_str: &str,
+        current_item: &Option<String>,
+        base_node: DefId,
+        extra_fragment: Option<String>,
+        ori_link: &str,
+        link_range: Option<Range<usize>>,
+    ) -> Option<(Res, Option<String>)> {
```

Comment:
This is getting a little unwieldy. Can you change it to take `CacheKey` instead?

Reply:
Ah there are worse things right now in my codebase locally. I'll clean those up (caching the resolution error went horribly wrong) and then we'll see what can be done here.

Reply:
Technically, this is the exact same signature as before, do you still want to take the `CacheKey`?